src/Microsoft.PowerShell.Security/security/CertificateProvider.cs

```diff
@@ -511,6 +496,55 @@ public bool Valid
             }
         }
 
+        private static CertificateStoreHandle CertOpenStore(CertStoreLocation location, string name, bool includeArchivedCerts)
+        {
+            SMASecurity.NativeMethods.CertOpenStoreFlags StoreFlags =
+                SMASecurity.NativeMethods.CertOpenStoreFlags.CERT_STORE_SHARE_STORE_FLAG |
+                SMASecurity.NativeMethods.CertOpenStoreFlags.CERT_STORE_SHARE_CONTEXT_FLAG |
+                SMASecurity.NativeMethods.CertOpenStoreFlags.CERT_STORE_OPEN_EXISTING_FLAG |
+                SMASecurity.NativeMethods.CertOpenStoreFlags.CERT_STORE_MAXIMUM_ALLOWED_FLAG;
+
+            if (includeArchivedCerts)
+            {
+                StoreFlags |= SMASecurity.NativeMethods.CertOpenStoreFlags.CERT_STORE_ENUM_ARCHIVED_FLAG;
+            }
+
+            switch (location)
+            {
+                case CertStoreLocation.LocalMachine:
+                    StoreFlags |= SMASecurity.NativeMethods.CertOpenStoreFlags.CERT_SYSTEM_STORE_LOCAL_MACHINE;
+                    break;
+
+                case CertStoreLocation.CurrentUser:
+                    StoreFlags |= SMASecurity.NativeMethods.CertOpenStoreFlags.CERT_SYSTEM_STORE_CURRENT_USER;
+                    break;
+
+                case CertStoreLocation.Service:
+                    StoreFlags |= SMASecurity.NativeMethods.CertOpenStoreFlags.CERT_SYSTEM_STORE_SERVICES;
+                    break;
+
+                default:
+                    // ThrowItemNotFound(storeLocation.ToString(), CertificateProviderItem.StoreLocation);
```

Comment:
Would it be better to throw?  The current code won't hit, but future changes may expose a bug if new changes occur.

Reply:
This was the existing behaviour, I have no idea if it was ignored by default but just kept it in case it was important.